### PR TITLE
[AIRFLOW-3391] Upgrade pendulum to latest major version.

### PR DESCRIPTION
**Pull request by jmcarp**
Make sure you have checked _all_ steps below.

### Jira

- [x] My PR addresses the following [Airflow Jira](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
  - https://issues.apache.org/jira/browse/AIRFLOW-3391
  - In case you are fixing a typo in the documentation you can prepend your commit with \[AIRFLOW-XXX\], code changes always need a Jira issue.

### Description

- [x] Here are some details about my PR, including screenshots of any UI changes:

We're a major version behind on pendulum; we should upgrade to >=2.0.

### Tests

- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

Covered by existing tests.

### Commits

- [x] My commits all reference Jira issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
  1. Subject is separated from body by a blank line
  1. Subject is limited to 50 characters (not including Jira issue reference)
  1. Subject does not end with a period
  1. Subject uses the imperative mood ("add", not "adding")
  1. Body wraps at 72 characters
  1. Body explains "what" and "why", not "how"

### Documentation

- [x] In case of new functionality, my PR adds documentation that describes how to use it.
  - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality

- [x] Passes `flake8`

### CODE REVIEW

File: tests/models.py [hunk 1]
```diff
@@ -230,9 +230,9 @@ def test_dag_topological_sort(self):
         self.assertEquals(tuple(), dag.topological_sort())
 
     def test_dag_naive_default_args_start_date(self):
-        dag = DAG('DAG', default_args={'start_date': datetime.datetime(2018, 1, 1)})
+        dag = DAG('DAG', default_args={'start_date': timezone.datetime(2018, 1, 1)})
```
Comment (bolkedebruin) [inline tests/models.py]: This is not naive?

File: tests/models.py [hunk 2]
```diff
@@ -230,9 +230,9 @@ def test_dag_topological_sort(self):
...
         self.assertEqual(dag.timezone, settings.TIMEZONE)
-        dag = DAG('DAG', start_date=datetime.datetime(2018, 1, 1))
+        dag = DAG('DAG', start_date=timezone.datetime(2018, 1, 1))
```
Comment (bolkedebruin) [inline tests/models.py]: Idem

**Review by bolkedebruin — CHANGES_REQUESTED** [review]
It's good to upgrade, but I think some changes are made that change the bahavior of Airflow with naive date times. Please keep the essence of the tests as is.

**Comment by Fokko** [comment]
There seems to be an issue with Travis, I've restarted the build.

**Comment by RosterIn** [comment]
Should I try to raise a PR to update to at least 1.5.1 or this PR / https://github.com/apache/airflow/pull/4507 will be fixed soon?

1.5.1 is the last version before bumping to 2.0
https://github.com/sdispater/pendulum/releases

**Comment by Fokko** [comment]
I believe @bolkedebruin is a bit busy. Feel free to pick it up @RosterIn

**Comment by pcockwell** [comment]
I am currently facing issues that this would fix. Very eager to see this merged. @jmcarp @bolkedebruin @Fokko Is there an ETA on when this might get approved/merged? At least that way I can plan accordingly for my use case.

**Comment by potiuk** [comment]
@pcockwell @jmcarp - I think it needs to be rebased first. There are some conflicts. I am happy to review the code afterwards (I have a bit more time now :) )

**Comment by galuszkak** [comment]
@potiuk I will rebase it and open new PR.

**Comment by galuszkak** [comment]
@potiuk I've rebased it in #5332

**Pull request closed without merging**